### ISSUE TITLE
[Windows] Snapshot executed twice

### ISSUE THREAD

**Issue by AlexV525** [dart-lang/pub#2875]
# Environment

- `pub version` or `flutter pub version`: Pub 2.13.0-0.0.dev
- OS version: Windows 10 (2004)
- Are you using the Chinese community mirror or a corporate firewall? **Yes**

# Problem

Continue with #2873 . Global activated snapshot will executed twice with only one command on Windows.

## Step to reproduce

<details>
<summary>Case 1: (flutter_asset_generator)</summary>

1. `dart pub global activate flutter_asset_generator`.
2. Find an example that has assets defined in pubspec. (e.g. https://github.com/CaiJingLong/flutter_resource_generator/tree/master/example)
3. `fgen --no-preview --no-watch`.

</details>

<details>
<summary>Case 2: (ff_annotation_route)</summary>

1. `dart pub global activate ff_annotation_route`.
2. Use the example that has routes annotated. (e.g. https://github.com/fluttercandies/ff_annotation_route/tree/master/example)
3. `ff_route`.

</details>

## Expected behavior

Executed only once.

![image](https://user-images.githubusercontent.com/15884415/107223690-756e7300-6a51-11eb-8bdc-dd526c8c7c82.png)

## Actual behavior

Executed twice.

![image](https://user-images.githubusercontent.com/15884415/107223806-9767f580-6a51-11eb-87f5-c17c8446678d.png)

![image](https://user-images.githubusercontent.com/15884415/107222787-515e6200-6a50-11eb-8bb9-edc259fb9782.png)

**Comment by sigurdm** [comment]
I was not able to reproduce
```
C:\Users\sigurdm\AppData\Local\Pub\Cache\bin\fgen  --no-preview --no-watch
Generating files for Project: C:\Users\sigurdm\flutter_resource_generator\example
format code 324ms
Generate dart resource file finish.
PS C:\Users\sigurdm\flutter_resource_generator\example> 
```

**Comment by AlexV525** [comment]
Okay, I tried with the cmd and Powershell again and it's fine. The twice execution might related to **Cmder**.

**Comment by AlexV525** [comment]
Also, directly run snapshot from cmd and terminal in IDE's (e.g. Android Studio) still taken over from the new script. I think at least a warning can be added to the dart website, in the pub section.

**Comment by sigurdm** [comment]
I think this is usual behavior of batch files - not sure we want to warn about strange behaviors of command interopreters.

**Comment by cezaryjk** [comment]
Hi gays,
Thanks a lot for your fruitful work! 

I'm afraid that this issue is still pending.
I'm on Pub 2.12.0
Executable installation on Windows 10 create batch file with risky GOTO command inside IF block. In some Windows version at least.
"...when GOTO is executed any active FOR/IF command is cancelled and the Batch file continue from the label on. – Aacini ..." 
( [https://stackoverflow.com/questions/8481558/windows-batch-goto-within-if-block-behaves-very-strangely](url)  )

My suggestion is to change batch file structure with something like that: 

`@echo off
rem This file was created by pub v2.12.0.
rem Package: fvm
rem Version: 1.3.7
rem Executable: fvm
rem Script: main
if exist "d:\src\flutter\.pub-cache\global_packages\fvm\bin\main.dart-2.12.0.snapshot" goto with_snapshot
goto nok_snapshot

:with_snapshot
  dart "d:\src\flutter\.pub-cache\global_packages\fvm\bin\main.dart-2.12.0.snapshot" %*
  rem The VM exits with code 253 if the snapshot version is out-of-date.
  rem If it is, we need to delete it and run "pub global" manually.
  if not errorlevel 253 goto snapshot_ok 
  goto nok_snapshot

:nok_snapshot
  pub global run fvm:main %*

goto eof

:error
exit /b %errorlevel%

:snapshot_ok

:eof
`
How do you think?

BR
Cezary
[fvmck.bat.txt](https://github.com/dart-lang/pub/files/6123215/fvmck.bat.txt)

**Comment by Levi-Lesches** [comment]
@cezaryjk I have this issue too, the problem was the line `pub global run fvm:main %*` was repeated. Deleting the first instance of it makes it work fine. I opened a new issue for visibility: #2934
